src/protocol.rs

```diff
@@ -339,9 +341,47 @@ impl ParticipantIdentifier {
     }
 }
 
-impl ProofContext for Vec<ParticipantIdentifier> {
+/// The SharedContext contains fixed known parameters accross the entire
+/// protocol. It does not however contain the entire protocol context.
+pub struct SharedContext {
+    pub participants: Vec<ParticipantIdentifier>,
+    pub generator: CurvePoint,
+    pub order: BigNumber,
```

Comment:
nit: Can these fields be made private? I don't see any direct access outside this module and I prefer to lean toward less-permissive internal APIs where possible.